gabbi/runner.py

```diff
@@ -59,6 +59,13 @@ def run():
 
     Output is formatted as unittest summary information.
     """
+    argv = sys.argv
+    try:  # extract file paths, separated by `--`
+        i = argv.index("--")
+        input_files = argv[i + 1:]
+        argv = argv[:i]
+    except ValueError:  # no file paths
+        input_files = [sys.stdin]
```

Comment:
I'd prefer you do this after the argparse argument declarations (or extract those to another method, call that and then argv munging).